src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileReader.kt

```diff
@@ -107,4 +111,24 @@ class CsvFileReader internal constructor(
         }
         return null
     }
+
+    /**
+     * deduplicate headers based on occurrence by appending "_<NUM>"
+     * Ex: [a,b,b,c,a] => [a,b,b_2,c,a_2]
+     *
+     * @return return headers as List<String>.
+     *
+     */
+    private fun deduplicateHeaders(headers: List<String>): List<String> {
```

Comment:
It seems that this logic cannot handle a CSV file's header `[a, a, a_1]`.
In this case, I think the auto-renamed header should be `[a, a_2, a_1]` (or `[a, a_1_1, a_1]`).

Reply:
That's a tricky case! Thanks for bringing this up, I'll put some thought into it.

Reply:
@JulianSudendorf 
Hi, if you don't mind, I could merge this PR and do this fix on my end.

Reply:
Hey @doyaaaaaken I'm very sorry for the delay. I experimented with a few options, but didn't like any of them completely. The most favorable option would be to introduce a new exception that indicates the naming collision, because automatically renaming the collision again can be very dependent o the use case. If you're comfortable merging the PR in the current state and handle the rest on your side I wouldn't mind at all :)

Reply:
> The most favorable option would be to introduce a new exception that indicates the naming collision, because automatically renaming the collision again can be very dependent o the use case.

Sounds good, I'll consider it. Thanks.

Reply:
I implemented like this. Thanks.
https://github.com/doyaaaaaken/kotlin-csv/pull/92/files#diff-63c02e69ac51828469fa44563f04f35ea00bc908a5af8343828a683a42bf11baR131-R132